packages/pigeon/platform_tests/alternate_language_test_plugin/android/src/test/java/com/example/alternate_language_test_plugin/AllDatatypesTest.java

```diff
@@ -168,7 +179,7 @@ public void hasValues() {
             .setBoolList(Arrays.asList(new Boolean[] {true, false}))
             .setDoubleList(Arrays.asList(new Double[] {0.5, 0.25, 1.5, 1.25}))
             .setIntList(Arrays.asList(new Long[] {1l, 2l, 3l, 4l}))
-            .setList(Arrays.asList(new int[] {1, 2, 3, 4}))
+            .setList(Arrays.asList(genericList))
```

Comment:
This change is because an inherent limitation in Java `equals` implementations is that `List` equality only works if each element in the list has a useful implementation of equals, and (as demonstrated in the new generator code) arrays don't have that. So having the generic list's value be a nested *array* makes equality not work. Fixing that would require a bunch of special code generation—we would have to do custom equality checks of all collections to call out to `Arrays.equals` for certain types—for a case that I'm skeptical will ever matter in practice, and is a common behavior of the language anyway.

Since the specific type here didn't actually matter, I opted for just changing to something with a useful `equals` so that the test will pass.